rmf_site_editor/src/site/deletion.rs

```diff
@@ -114,14 +115,73 @@ impl Plugin for DeletionPlugin {
     }
 }
 
-fn handle_deletion_requests(mut deletions: EventReader<Delete>, mut params: DeletionParams) {
+pub struct DeletionBox {
+    system: BoxedSystem<HashSet<Entity>, HashSet<Entity>>,
+    initialized: bool,
+}
+
+impl DeletionBox {
+    pub fn new(system: BoxedSystem<HashSet<Entity>, HashSet<Entity>>) -> Self {
+        Self {
+            system: system,
+            initialized: false
+        }
+    }
+}
+
+#[derive(Default, Resource)]
+pub struct DeletionFilter {
+    boxed_systems: Vec<DeletionBox>,
+    pending_delete: HashSet<Delete>,
+}
```

Comment:
The struct fields are private by default and this resource has no public API so it won't be possible for users outside of this crate (or even file I believe?) to add their own deletion filters.